jmdaemon/jmdaemon/onionmc.py

```diff
@@ -0,0 +1,1210 @@
+        # each other (i.e. they are *required* to only configure
+        # themselves, not other dns). But this could happen by
+        # accident.
+        encoded_msg = OnionCustomMessage(pubmsg, msgtype)
+        for peer in self.get_connected_nondirectory_peers():
+            # don't loop back to the sender:
+            if peer.nick == from_nick:
+                continue
+            log.debug("Sending {}:{} to nondir peer {}".format(
+                msgtype, pubmsg, peer.peer_location()))
+            self._send(peer, encoded_msg)
+
+    def forward_privmsg_to_peer(self, nick: str, message: str,
+                                from_nick: str) -> None:
+        assert self.self_as_peer.directory
+        peer = self.get_peer_by_nick(nick)
+        if not peer:
+            log.debug("We were asked to send a message from {} to {}, "
+                      "but {} is not connected.".format(from_nick, nick, nick))
+            return
+        # The `message` passed in has format COMMAND_PREFIX||command||" "||msg
+        # we need to parse out cmd, message for sending.
+        _, cmdmsg = message.split(COMMAND_PREFIX)
+        cmdmsglist = cmdmsg.split(" ")
+        cmd = cmdmsglist[0]
+        msg = " ".join(cmdmsglist[1:])
+        privmsg = self.get_privmsg(nick, cmd, msg, source_nick=from_nick)
+        #log.debug("Sending out privmsg: {} to peer: {}".format(privmsg, peerid))
+        encoded_msg = OnionCustomMessage(privmsg,
+                        JM_MESSAGE_TYPES["privmsg"])
+        self._send(peer, encoded_msg)
+        # If possible, we forward the from-nick's network location
+        # to the to-nick peer, so they can just talk directly next time.
+        peer_from = self.get_peer_by_nick(from_nick)
+        if not peer_from:
+            return
+        self.send_peers(peer, peer_filter=[peer_from])
+
+    def process_control_message(self, peerid: str, msgtype: int,
+                                msgval: str) -> bool:
+        """ Triggered by a directory node feeding us
+        peers, or by a connect/disconnect hook; this is our housekeeping
+        to try to create, and keep track of, useful connections.
+        """
+        all_ctrl = list(LOCAL_CONTROL_MESSAGE_TYPES.values(
+            )) + list(CONTROL_MESSAGE_TYPES.values())
+        if msgtype not in all_ctrl:
+            return False
+        # this is too noisy, but TODO, investigate allowing
+        # some kind of control message monitoring e.g. default-off
+        # log-to-file (we don't currently have a 'TRACE' level debug).
+        #log.debug("received control message: {},{}".format(msgtype, msgval))
+        if msgtype == CONTROL_MESSAGE_TYPES["peerlist"]:
+            # This is the base method of seeding connections;
+            # a directory node can send this any time. We may well
+            # need to control this; for now it just gets processed,
+            # whereever it came from:

```

Comment:
```suggestion
            # wherever it came from:
```

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/bc05a929963f39aae55507af80123eb837a065e9